src/components/gas/GasSpeedButton.js

```diff
@@ -0,0 +1,583 @@
+        }
+        letterSpacing="roundedTight"
+        lineHeight="normal"
+        size="lmedium"
+        weight="bold"
+      >
+        {gasIsNotReady ? 'Loading...' : animatedNumber}
+      </Text>
+    ),
+    [theme, colors, gasIsNotReady]
+  );
+
+  const handlePressSpeedOption = useCallback(
+    selectedSpeed => {
+      if (
+        selectedSpeed === gasUtils.CUSTOM &&
+        isEmpty(gasFeeParamsBySpeed[gasUtils.CUSTOM])
+      ) {
+        const gasFeeParams = gasFeeParamsBySpeed[selectedGasFeeOption];
+        updateToCustomGasFee({
+          ...gasFeeParams,
+          maxBaseFeePerGas: gasFeeParams.maxFeePerGas,
+          maxPriorityFeePerGas: gasFeeParams.maxPriorityFeePerGas,
+          option: gasUtils.CUSTOM,
+        });
+      } else {
+        updateGasFeeOption(selectedSpeed);
+      }
+    },
+    [
+      gasFeeParamsBySpeed,
+      selectedGasFeeOption,
+      updateToCustomGasFee,
+      updateGasFeeOption,
+    ]
+  );
+
+  const formatTransactionTime = useCallback(() => {
+    const time = parseFloat(estimatedTimeValue || 0).toFixed(0);
+    let selectedGasFeeGwei = get(
+      selectedGasFee,
+      'estimatedFee.value.display.display'
+    );
+    if (selectedGasFeeGwei === '0 Gwei') {
+      selectedGasFeeGwei = '< 1 Gwei';
+    }
+    let timeSymbol = '~';
+
+    if (selectedGasFeeOption === CUSTOM) {
+      const customWei = get(
+        gasFeesBySpeed,
+        `${CUSTOM}.estimatedFee.value.amount`
+      );
+      if (customWei) {
+        const normalWei = get(
+          gasFeesBySpeed,
+          `${NORMAL}.estimatedFee.value.amount`
+        );
+        const urgentWei = get(
+          gasFeesBySpeed,
+          `${URGENT}.estimatedFee.value.amount`
+        );
+        const minGasPriceSlow = normalWei | urgentWei;
+        const maxGasPriceFast = urgentWei;
+        if (normalWei < minGasPriceSlow) {
+          timeSymbol = '>';
+        } else if (normalWei > maxGasPriceFast) {
+          timeSymbol = '<';
+        }
+
+        return ` ${timeSymbol}${time} ${estimatedTimeUnit}`;
+      } else {
+        return ``;
+      }
+    }
+
+    // If it's still loading show `...`
+    if (time === '0' && estimatedTimeUnit === 'min') {
+      return ``;
+    }
+
+    return ` ${timeSymbol}${time} ${estimatedTimeUnit}`;
+  }, [
+    estimatedTimeUnit,
+    estimatedTimeValue,
+    gasFeesBySpeed,
+    selectedGasFee,
+    selectedGasFeeOption,
+  ]);
+
+  const openGasHelper = useCallback(() => {
+    android && Keyboard.dismiss();
+    navigate(Routes.EXPLAIN_SHEET, { type: 'gas' });
+  }, [navigate]);
+
+  const handlePressMenuItem = useCallback(
+    ({ nativeEvent: { actionKey } }) => {
+      handlePressSpeedOption(actionKey);
+    },
+    [handlePressSpeedOption]
+  );
+
+  const nativeFeeCurrencySymbol = useMemo(() => {
+    switch (currentNetwork) {
+      case networkTypes.polygon:
+        return { address: MATIC_POLYGON_ADDRESS, symbol: 'MATIC' };
+      case networkTypes.optimism:
```

Comment:
Depending on which PR gets in first, may need to also update for Optimism v2 asset info